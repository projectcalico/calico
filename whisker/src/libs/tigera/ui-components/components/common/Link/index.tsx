import * as React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import { Link as ChakraLink, LinkProps, forwardRef } from '@chakra-ui/react';

const Link: React.FC<React.PropsWithChildren<LinkProps & { to?: string }>> =
    forwardRef(({ href, to, ...rest }, ref) => (
        <ChakraLink
            ref={ref}
            {...(href && { href })}
            {...(to && { as: ReactRouterLink, to })}
            tabIndex={0}
            {...rest}
        />
    ));

export default Link;
