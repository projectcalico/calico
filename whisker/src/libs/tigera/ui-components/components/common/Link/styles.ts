export default {
    baseStyle: {
        fontStyle: 'normal',
        fontWeight: 'semibold',
        fontSize: 'xxs',
        lineHeight: '4',
        color: 'tigera-color-primary',
        svg: {
            fill: 'tigera-color-primary',
            color: 'tigera-color-primary',
        },
        _hover: {
            color: 'tigeraBlueMedium',
            textDecoration: 'none',
            svg: {
                fill: 'tigeraBlueMedium',
                color: 'tigeraBlueMedium',
            },
        },
        _focus: {
            boxShadow: 'none',
        },
        ':focus:not(:focus-visible)': {
            outline: 'none',
        },
        _dark: {
            _hover: {
                color: 'tigeraGoldMedium40',
                textDecoration: 'none',
                svg: {
                    fill: 'tigeraGoldMedium40',
                    color: 'tigeraGoldMedium40',
                },
            },
        },
    },
    variants: {
        underlined: {
            fontSize: 'xs',
            fontWeight: '500',
            _hover: {
                textDecoration: 'underline',
            },
            _dark: {
                _hover: {
                    textDecoration: 'underline',
                },
            },
        },
    },
};
