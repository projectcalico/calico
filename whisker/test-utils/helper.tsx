import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
