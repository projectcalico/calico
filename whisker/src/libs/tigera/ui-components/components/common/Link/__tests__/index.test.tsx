import { renderWithRouter } from '@/test-utils/helper';
import Link from '../index';

describe('<Link/>', () => {
    it('it renders with to (internal url)', () => {
        const { asFragment } = renderWithRouter(<Link to='/test' />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('it renders with href', () => {
        const { asFragment } = renderWithRouter(
            <Link href='http://tigera.io/' />,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
