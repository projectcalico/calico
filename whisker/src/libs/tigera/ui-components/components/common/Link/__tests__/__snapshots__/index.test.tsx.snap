// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link/> it renders with href 1`] = `
<DocumentFragment>
  <a
    class="chakra-link css-1mpq44i"
    href="http://tigera.io/"
    tabindex="0"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Link/> it renders with to (internal url) 1`] = `
<DocumentFragment>
  <a
    class="chakra-link css-1mpq44i"
    href="/test"
    tabindex="0"
  />
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
